File: sacapromedio.go
package main

import (
	"fmt"
)

func main() {
	var nota1, nota2, nota3 float64

	// Capturar las notas del usuario
	fmt.Print("Ingrese nota 1: ")
	fmt.Scan(&nota1)

	fmt.Print("Ingrese nota 2: ")
	fmt.Scan(&nota2)

	fmt.Print("Ingrese nota 3: ")
	fmt.Scan(&nota3)

	// Calcular la suma y el promedio
	suma := nota1 + nota2 + nota3
	promedio := suma / 3

	// Determinar el estado
	estado := "REPROBADO"
	if promedio >= 4.0 {
		estado = "APROBADO"
	}

	// Mostrar resultados
	fmt.Println("\n------ Resultados ----")
	fmt.Printf("Promedio: %.2f\n", promedio)
	fmt.Println("Estado:", estado)
}

File: sacapromedio.ts
import * as readlineSync from "readline-sync";

// Función para capturar y convertir la entrada del usuario
function obtenerNota(mensaje: string): number {
    let nota = parseFloat(readlineSync.question(mensaje + " "));
    if (isNaN(nota)) {
        console.log("Entrada no válida. Inténtalo de nuevo.");
        return obtenerNota(mensaje);
    }
    return nota;
}

// Capturar las notas del usuario
const nota1: number = obtenerNota("Ingrese nota 1:");
const nota2: number = obtenerNota("Ingrese nota 2:");
const nota3: number = obtenerNota("Ingrese nota 3:");

// Calcular la suma y el promedio
const suma: number = nota1 + nota2 + nota3;
const promedio: number = suma / 3.0;

// Determinar el estado
const estado: string = promedio >= 4.0 ? "APROBADO" : "REPROBADO";

// Mostrar resultados
console.log("\n------ Resultados ----");
console.log(`Promedio: ${promedio.toFixed(2)}`);
console.log(`Estado: ${estado}`);

